components/automate-ui/src/app/components/page-picker/page-picker.component.ts

```diff
@@ -56,4 +89,39 @@ export class PagePickerComponent implements OnChanges {
     return item;
   }
 
+  handleSelectItem(event, value): void {
+    if (event.isUserInput) {
+      this.pageChanged.emit(value);
+    }
+  }
+
+  handleSelectMaxPageItems(event, value): void {
+    if (event.isUserInput) {
+      const updatedPageNumber = this.getUpdatedPageNumber(value);
+
+      this.pageSizeChanged.emit({
+        pageSize: value,
+        updatedPageNumber
+      });
+    }
+  }
+
+  private getUpdatedPageNumber(value): number {
+    let newPageNumber;
+    const newTotalPages = Math.ceil(this.total / value);
+
+    // handle first page
+    if (this.itemStartCount === 1) {
+      newPageNumber = 1;
+    // handle last page
+    } else if (this.itemStartCount < this.total && (this.itemStartCount > (this.total - value))) {
+      newPageNumber = newTotalPages;
+    // handle middle pages
+    } else {
+      newPageNumber = Math.ceil(this.itemStartCount / value);
+    }
+
+    return newPageNumber;
+  }
```

Comment:
After a user selects a new amount of items that should be displayed per page, this function determines what page the user should be on.
i.e. user is on page 11, viewing 101 - 110, 10 items per page.
user selects to view 50 items per page.
user is now on page 3, viewing 101 - 150